pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -438,6 +438,21 @@ func (s *Reconciler) Delete(ctx context.Context) error {
 		return fmt.Errorf("failed to delete OS disk: %w", err)
 	}
 
+	for _, dd := range s.scope.MachineConfig.DataDisks {
```

Comment:
I have doubts about non-conditional deeltion of datadisks. Can we make it configurable?

Reply:
Noting from discussion today: this should use the most severe admonition we have - very important to avoid data loss